src/Hilke.KineticConvolution/DoubleAlgebraicNumber/DoubleConverter.cs

```diff
@@ -0,0 +1,252 @@
+        {
+            if (direction is null)
+            {
+                throw new ArgumentNullException(nameof(direction));
+            }
+
+            return AlgebraicNumberFactory.CreateDirection(FromDouble(direction.X), FromDouble(direction.Y));
+        }
+
+        public virtual Direction<double> ToDouble(Direction<TAlgebraicNumber> direction)
+        {
+            if (direction is null)
+            {
+                throw new ArgumentNullException(nameof(direction));
+            }
+
+            return DoubleFactory.CreateDirection(ToDouble(direction.X), ToDouble(direction.Y));
+        }
+
+        public virtual DirectionRange<TAlgebraicNumber> FromDouble(DirectionRange<double> range)
+        {
+            if (range is null)
+            {
+                throw new ArgumentNullException(nameof(range));
+            }
+
+            return AlgebraicNumberFactory.CreateDirectionRange(
+                FromDouble(range.Start),
+                FromDouble(range.End),
+                range.Orientation);
+        }
+
+        public virtual DirectionRange<double>? ToDouble(DirectionRange<TAlgebraicNumber> range)
+        {
+            if (range is null)
+            {
+                throw new ArgumentNullException(nameof(range));
+            }
+
+            var start = ToDouble(range.Start);
+            var end = ToDouble(range.End);
+            
+            return range.Start != range.End && start == end
+                       ? resolveLimitCase()
+                       : DoubleFactory.CreateDirectionRange(start, end, range.Orientation);
+
+            DirectionRange<double>? resolveLimitCase()
+            {
+                var referenceDirection = range.Start.Opposite();
+
+                var result = range.Start.CompareTo(range.End, referenceDirection) == DirectionOrder.Before
+                                 ? range.Orientation == Orientation.CounterClockwise
+                                       ? null
+                                       : DoubleFactory.CreateDirectionRange(start, end, range.Orientation)
+                                 : range.Orientation == Orientation.CounterClockwise
+                                     ? DoubleFactory.CreateDirectionRange(start, end, range.Orientation)
+                                     : null;
+
+                return result
+                    ?? Mode switch
+                       {
+                           InvalidityManagementMode.Silent => null,
+                           InvalidityManagementMode.ThrowException =>
+                               throw new InvalidOperationException(
+                                   "The range collapsed to a single direction during conversion to double."),
+                           _ => throw new NotSupportedException(
+                                    $"Only Silent and ThrowException modes are supported but got {Mode.GetType()}.")
+                       };
```

Comment:
The logic inthere is quite involved. Couldn't we get away with something like
```csharp
if (range.Start != range.End && start == end)
{
    return DoubleFactory.CreateDirectionRange(start, end, range.Orientation);
}

switch (Mode)
{
    case InvalidityManagmentMode.Silent: return null;
    case InvalidityManagmentMode.ThrowException: throw new InvalidOperationException(...);
    default: throw new NotSupportedException($"Only {nameof(InvalidityManagmentMode.Silent)} and {nameof(InvalidityManagmentMode.ThrowException)} invalid conversion policies are supported.");
}
```
?

Reply:
Much better